### PR TITLE
driver: gpio: Add pin_configure api for creg_gpio driver

### PR DESCRIPTION
**Pull request by SiyuanCheng-CN**
Update pin_configure api for creg_gpio driver.

### CODE REVIEW

File: drivers/gpio/gpio_creg_gpio.c [hunk 1]
```diff
@@ -119,6 +112,50 @@ static int pin_interrupt_configure(const struct device *dev,
 	return -ENOTSUP;
 }
 
+static int pin_config(const struct device *dev,
```
Comment (tbursztyka) [inline drivers/gpio/gpio_creg_gpio.c:115]: I'd split up this patch in 2, first being improving the creg_gpio driver (so far, pin_config() implementation is the driver improvement. The spi cs is just the use case that triggered the requirement for such improvement).

And second patch for the em_starterkit max freq. (In the commit message, explain why you reduced such freq btw)

**Review by tbursztyka — DISMISSED** [review]

File: drivers/gpio/gpio_creg_gpio.c (hunk 1 above)
Comment (henrikbrixandersen) [reply drivers/gpio/gpio_creg_gpio.c:115]: Agreed.

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

File: drivers/gpio/gpio_creg_gpio.c (hunk 1 above)
Comment (SiyuanCheng-CN) [reply drivers/gpio/gpio_creg_gpio.c:115]: Thanks for the suggestion, now updated.

**Review by ruuddw — APPROVED** [review]

**Review by cfriedt — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Comment by henrikbrixandersen** [comment]
@tbursztyka ping

**Pull request merged**